sdk/core/Azure.Core/src/Shared/EventSourceEventFormatting.cs

```diff
@@ -18,12 +19,24 @@ public static string Format(EventWrittenEventArgs eventData)
 
             if (eventData.Message != null)
             {
-                return string.Format(CultureInfo.InvariantCulture, eventData.Message, payloadArray);
+                try
+                {
+                    return string.Format(CultureInfo.InvariantCulture, eventData.Message, payloadArray);
+                }
+                catch (FormatException)
+                {
```

Comment:
We should consider adding a line here to call out that the message could not be formatted.   If we get to this point, I believe that it indicates a bug in our logging code and it would be helpful if those that encounter this have the awareness and information to open a bug.

Reply:
Yeah, we should also think if we can catch it earlier, some analyzer or test utility possibly.